### DISCUSSION TITLE
Possible inaccurate information in the vcpkg build doc leading to build error

### DISCUSSION THREAD

**Discussion by finalreq** [microsoft/cpp_client_telemetry#1056] [Q&A]
I am trying to get a vcpkg build to integrate 1DS into our code base and the [build instruction documentation](https://github.com/microsoft/cpp_client_telemetry/blob/master/docs/building-with-vcpkg.md) for vcpkg is not accurate. It says mstelemetry is already in vcpkg which as far as I can tell is not. But the main issue is in #810 @maxgolov left a comment requesting the documentation to be updated and it seems that never happened. Currently I'm trying to get a build for windows and I encounter an error that says "error MSB4126: The specified solution configuration "Release|x86" is invalid.". But when I look at the command ran, configuration is specified as Release and not Release|x86. see below for full command printed to cmd.

(msbuild D:/vcpkg/buildtrees/mstelemetry/x86-windows-rel/275154d47f-c88f9d4ac2.clean/Solutions/MSTelemetrySDK.sln /p:Configuration=Release /p:MATSDK_SHARED_LIB=1 /t:sqlite:Rebuild,win32-lib:Rebuild /p:Platform=x86 /p:PlatformToolset=v142 /p:VCPkgLocalAppDataDisabled=true /p:UseIntelMKL=No /p:WindowsTargetPlatformVersion=10.0.22000.0 /p:VcpkgTriplet=x86-windows /p:VcpkgInstalledDir=D:/vcpkg/installed /p:VcpkgManifestInstall=false /p:UseMultiToolTask=true /p:MultiProcMaxCount=13 /p:EnforceProcessCountAcrossBuilds=true /m:13 /p:WholeProgramOptimization=false /p:ForceImportBeforeCppTargets=D:/vcpkg/scripts/buildsystems/msbuild/vcpkg.targets /p:VcpkgApplocalDeps=false)

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_msbuild.cmake:79 (vcpkg_execute_required_process)
  D:/oneds/cpp_client_telemetry/tools/ports/mstelemetry/portfile.cmake:65 (vcpkg_install_msbuild)
  scripts/ports.cmake:147 (include)

Any guidance as to the cause of this error and how to fix it? I'm assuming that I'm missing something that perhaps wasn't clarified in the build docs but please let me know. I've just followed the steps in the documentation I linked above, and it led to that error. I'm running the lates repo pull and building on win 11.

**Comment by lalitb** [comment]
@finalreq  

> It says mstelemetry is already in vcpkg which as far as I can tell is not.

I don't see documentation stating that. The mstelemetry is not available on public vcpkg. It's vcpkg port is available in the repo, but not distributed publically through vcpkg repositories.

> Any guidance as to the cause of this error and how to fix it?

Can you try setting tripet to correct architecture, and try again. As an example, these command works for me from the repo root folder.

PS> $env:VCPKG_DEFAULT_TRIPLET = 'x64-windows'
PS> .\tools\vcpkg\vcpkg.exe install --head --overlay-ports=.\tools\ports\ mstelemetry

**Reply by finalreq** [reply]
@lalitb Thank you for the reply. Reading through guidance for adding open source code to OS repo projects, it seems they require the project to be on the public vcpkg and mirrored internally. Would it be possible to do this or was mstelemetry never meant to be on the public vcpkg?

**Reply by maxgolov** [reply]
No, it doesn't have to be public.

**Reply by finalreq** [reply]
https://www.osgwiki.com/wiki/Using_OSS_in_the_OS_Repo#2._Verify_the_OSS_Project_has_a_VCPkg_Recipe
According to this it needs to be in vcpkg and accroding to step 3 it needs to be mirrored internally from vcpkg.

**Comment by maxgolov** [comment]
f.y.i. - we discussed it earlier in the day with @finalreq offline that yet another solution is to build as
`.\tools\vcpkg\vcpkg.exe install --head --overlay-ports=.\tools\ports\ mstelemetry:x64-windows`
It seems like it worked fine. It's the same as what Lalit suggested above.
